File: doordeck-sdk/src/mingwMain/resources/csharp/Model/Responses/SiteResponses.cs
namespace Doordeck.Headless.Sdk.Model.Responses;

public class SiteResponse
{
    public required Guid Id { get; set; }
    public required string Name { get; set; }
    public required string Colour { get; set; }
    public required double Longitude { get; set; }
    public required double Latitude { get; set; }
    public required int Radius { get; set; }
    public required DateTime Created { get; set; }
    public required DateTime Updated { get; set; }
}

public class UserForSiteResponse
{
    public required Guid UserId { get; set; }
    public required string Email { get; set; }
    public string? DisplayName { get; set; }
    public required bool Orphan { get; set; }
}

File: library/src/commonMain/kotlin/com/doordeck/multiplatform/sdk/api/responses/SiteResponses.kt
package com.doordeck.multiplatform.sdk.api.responses

import com.doordeck.multiplatform.sdk.api.model.UserRole
import kotlinx.serialization.Serializable
import kotlin.js.JsExport

@JsExport
@Serializable
class SiteResponse(
    val id: String,
    val name: String,
    val colour: String,
    val longitude: Double,
    val latitude: Double,
    val radius: Int,
    val passBackground: String,
    val created: String,
    val updated: String
)

@JsExport
@Serializable
class SiteLocksResponse(
    val id: String,
    val name: String,
    val colour: String? = null,
    val role: UserRole,
    val settings: SiteLockSettingsResponse
)

@JsExport
@Serializable
class SiteLockSettingsResponse(
    val unlockTime: Double,
    val permittedAddresses: Array<String>,
    val defaultName: String,
    val tiles: Array<String>,
    val state: SiteStateResponse? = null,
    val favourite: Boolean? = null
)

@JsExport
@Serializable
class SiteStateResponse(
    val connected: Boolean
)

@JsExport
@Serializable
class UserForSiteResponse(
    val userId: String,
    val email: String,
    val displayName: String? = null,
    val orphan: Boolean
)
